src/cache.rs

```diff
@@ -270,33 +422,40 @@ impl<T: Hash + PartialEq> Constraint<T> {
     where
         F: FnMut(&T) -> u128,
     {
-        let calls = self.0.borrow();
-        ACCELERATOR.with(|accelerator| {
-            let mut map = accelerator.borrow_mut();
-            calls.iter().all(|entry| {
-                *map.entry((id, entry.both)).or_insert_with(|| f(&entry.args))
-                    == entry.ret
+        let accelerator = accelerator(id);
+        let inner = self.0.read();
+        if let Some(accelerator) = accelerator {
+            let mut map = accelerator.lock();
+            inner.values().all(|entry| {
+                *map.entry(entry.both).or_insert_with(|| f(&entry.args)) == entry.ret
             })
-        })
+        } else {
+            inner.values().all(|entry| f(&entry.args) == entry.ret)
```

Comment:
I'm a bit afraid of code bloat (through many inline instantiations) here, but it's probably ok.

Reply:
The reason I did it this way is because otherwise I would need to panic (since this is the case where a `Tracked` outlives the eviction, which in multithreaded code may very well occur).